backend/utils.py

```
@@ -321,7 +327,7 @@ def generate_reference_id() -> str:
     import random
     import string
     from datetime import datetime
-    
-    timestamp = datetime.now().strftime('%Y%m%d-%H%M%S')
-    random_suffix = ''.join(random.choices(string.ascii_uppercase + string.digits, k=4))
+
+    timestamp = datetime.now().strftime("%Y%m%d-%H%M%S")
+    random_suffix = "".join(random.choices(string.ascii_uppercase + string.digits, k=4))
```

Ruff (0.15.12)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)